demos/android/MASVS-RESILIENCE/MASTG-DEMO-0039/MASTG-DEMO-0039.md

```diff
@@ -4,6 +4,31 @@ title: Detecting StrictMode PenaltyLog Usage with Semgrep
 id: MASTG-DEMO-0039
 code: [kotlin]
 test: MASTG-TEST-0265
-status: draft
-note: This demo shows how to detect the use of StrictMode in the codebase using Semgrep.
+status: new
 ---
+
+### Sample
+
+This sample demonstrates the detection of `StrictMode` uses at runtime using Frida. The app enables a `StrictMode` policy to detect leaked SQLite objects and intentionally leaves a cursor unclosed to trigger the policy.
+
+{{ ../MASTG-DEMO-0037/MastgTest.kt # MastgTest_reversed.java }}
+
+### Steps
+
+Let's run @MASTG-TOOL-0110 rules against the sample code.
+
+{{ ../../../../rules/mastg-android-strictmode.yml }}
+
+{{ run.sh }}
```

Comment:
Isn't there a step missing? Since you run the semgrep rule against java code, I guess the app needs to be decompiled?

But since decompiling does not work so reliably, why not run a semgrep rule against the smali code?

Reply:
You're right. However, we have many demos following exactly this pattern. I'll leave it like it is for now because it's gonna be easier to just replace them all at once.